OsmAnd/src/net/osmand/plus/wikivoyage/data/TravelArticle.java

```diff
@@ -163,6 +177,101 @@ public static String getImageUrl(@NonNull String imageTitle, boolean thumbnail)
 		return IMAGE_ROOT_URL + "thumb/" + hash[0] + "/" + hash[1] + "/" + imageTitle + "/" + prefix + imageTitle + suffix;
 	}
 
+	@Nullable
+	public GPXFile buildGpxFile(@NonNull List<BinaryMapIndexReader> readers){
+		GPXFile gpxFile = null;
+		List<Amenity> pointList = getPointList(readers);
+		if (!Algorithms.isEmpty(pointList)) {
+			gpxFile = new GPXFile(getTitle(), getLang(), getContent());
+			gpxFile.metadata.link = TravelArticle.getImageUrl(getImageTitle(), false);
+			for (Amenity amenity : pointList) {
+				WptPt wptPt = createWptPt(amenity, getLang());
+				gpxFile.addPoint(wptPt);
+			}
+		}
+		return gpxFile;
+	}
+
+	@NonNull
+	private synchronized List<Amenity> getPointList(List<BinaryMapIndexReader> readers) {
```

Comment:
this method should not be synchronized !!! Otherwise all methods should be synchronized!

Reply:
It should be part of TravelHelper method.

Synchronized is not clear needed or not? Especially private method!!!